RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc

```diff
@@ -104,36 +34,32 @@ reco::Candidate::Point PositionAtECalEntranceComputer::operator()(const reco::Ca
                                                                   bool& success) const {
   assert(bField_z_ != -1.);
   reco::Candidate::Point position;
+  BaseParticlePropagator propagator = BaseParticlePropagator(
+      RawParticle(particle->p4(),
+                  math::XYZTLorentzVector(particle->vertex().x(), particle->vertex().y(), particle->vertex().z(), 0.),
+                  particle->charge()),
+      0.,
+      0.,
+      bField_z_);
   if (!isPhase2_ || std::abs(particle->eta()) < ecalBarrelEndcapEtaBorder_) {  // ECal
-    BaseParticlePropagator propagator = BaseParticlePropagator(
-        RawParticle(particle->p4(),
-                    math::XYZTLorentzVector(particle->vertex().x(), particle->vertex().y(), particle->vertex().z(), 0.),
-                    particle->charge()),
-        0.,
-        0.,
-        bField_z_);
     propagator.propagateToEcalEntrance(false);
-    if (propagator.getSuccess() != 0) {
-      position = propagator.particle().vertex().Vect();
-      success = true;
-    } else {
+  } else {  // HGCal
+    if (std::abs(particle->vertex().z()) >= hgcalFace_z_) {
       success = false;
+      return position;
     }
-  } else {  // HGCal
+    propagator.setPropagationConditions(152.6, hgcalFace_z_, false);
+    propagator.propagate();
+  }
+  if (propagator.getSuccess() != 0) {
+    position = propagator.particle().vertex().Vect();
+    if (std::abs(position.eta()) > hgcalHfEtaBorder_) {
```

Comment:
simplify: `success = std::abs(position.eta()) <= hgcalHfEtaBorder_;`

Reply:
Can do this, but I think it makes code more difficult to read with very small, if any, gain (I suppose that compiler optimisation does good job in such cases).

Reply:
It's mainly to reduce verbosity: `if(true) x = true; else x = false` is unnecessarily repetitive.

Reply:
OK, changed.